contracts/test/core/UntronCore.setProvider.t.sol

```diff
@@ -534,19 +533,44 @@ contract SetProviderTest is UntronCoreBase {
         mintUSDT(providerAddress, liquidity);
         approveUSDT(providerAddress, address(untron), liquidity);
 
-        // When: The provider sets their profile with random valid inputs
+        // Ensure the initial state of the provider is as expected
+        IUntronCore.Provider memory initialProvider = untron.providers(providerAddress);
+        assertEq(initialProvider.liquidity, 0, "Initial provider's liquidity should be 0");
+        assertEq(initialProvider.rate, 0, "Initial provider's rate should be 0");
+        assertEq(initialProvider.minOrderSize, 0, "Initial provider's minOrderSize should be 0");
+        assertEq(initialProvider.minDeposit, 0, "Initial provider's minDeposit should be 0");
+
+        // Start the prank as the provider
         vm.startPrank(providerAddress);
-        untron.setProvider(liquidity, rate, minOrderSize, minDeposit, receivers);
-        vm.stopPrank();
 
-        // Then: Verify provider's information
-        IUntronCore.Provider memory provider = untron.providers(providerAddress);
-        assertEq(provider.liquidity, liquidity, "Provider's liquidity should match");
-        assertEq(provider.rate, rate, "Provider's rate should match");
-        assertEq(provider.minOrderSize, minOrderSize, "Provider's minOrderSize should match");
-        assertEq(provider.minDeposit, minDeposit, "Provider's minDeposit should match");
+        if (liquidity == 0) {
```

Comment:
Instead of doing this, create separate tests for each case
So we will have the existing one (fixed)
`testSetProvider_RevertIf_ZeroLiquidity`

Then we will also need to add
`testSetProvider_RevertIf_ZeroMinDeposit`
`testSetProvider_RevertIf_ZeroMinOrderSize`
`testSetProvider_RevertIf_ZeroRate`

Reply:
Noted!